internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -100,6 +110,135 @@ func validateSidecarMountPoints(in []manifest.SidecarMountPoint) error {
 	return nil
 }
 
+func validateSidecarDependsOn(in manifest.SidecarConfig, sidecarName string, sidecars map[string]*manifest.SidecarConfig, manifestName string) error {
+	if in.DependsOn == nil {
+		return nil
+	}
+
+	for name, status := range in.DependsOn {
+		// don't let a sidecar depend on itself
+		if name == sidecarName {
+			return errCircularDependency
+		}
+		// status must be one of < start | complete | success >
+		if !isValidStatus(status) {
+			return errInvalidDependsOnStatus
+		}
+		// essential containers must have 'start' as a status
+		if name == manifestName || sidecars[name].Essential == nil || *sidecars[name].Essential {
```

Comment:
```suggestion
		if name == manifestName || sidecars[name].Essential == nil || aws.BoolValue(sidecars[name].Essential) {
```
Reason is it would make it clearer that `Essential` is a `*bool` type.

Reply:
These AWS functions are also great for safe pointer dereference; `aws.BoolValue` returns false if the pointer is nil, letting you differentiate with an `== nil` check and no worry about panics.

Reply:
Adjusted